pinterest/teletraan
# Let the fields that are readonly in the Cluster Config > Advanced Settings page be configurable

Previously we had hard-coded checks to ensure that `spiffe_id` was readonly, and could not be changed by the user.

For context "User Data Configs" section would render like this, so spiffe couldn't be edited or removed:
<img width="512" alt="Screen Shot 2021-11-15 at 2 45 19 PM" src="https://user-images.githubusercontent.com/187460/141864584-7cf095cb-bcd9-4fc6-8d9a-9adf61b4bff4.png">

This makes the list of readonly fields a configuration setting, instead of being a hard-coded value. 
By making it a configuration we can create more readonly fields that will behave the same way that spiffe_id does.

Enables internal config change: D812125 which is in support of: D809778

---

**Test updates to an existing Cluster**
1. Run teletraan and navigate to an existing cluster which has a spiffe_id. Any internal cluster will have one.
2. View the `/env/$ENV/$STAGE/config/cluster/config/` path, and click on "Advanced Settings"
  - notice the `spiffe_id` textfield is readonly
3. Try to submit a different `spiffe_id`
   - You can try to edit the html, submit your own POST request outside the browser with curl...
   - Updates to the existing `spiffe_id` user-data value should be rejected.

**Test creating a new Cluster form**
1. Visit the new-cluster page, maybe for a new Stage
2. Create a user-data field with the key `spiffe_id` and any value
3. Try to submit this user-data
  - User-data with the key `spiffe_id` should be rejected no matter how you submit it.
